Vapor 2 support

vzsg opened the pull request:
This is the continuation of #26.
Resolves #25.

natebird committed:
Update dependencies to 2.0 betas.

natebird committed:
Update travis and circle to use swift 3.1

natebird committed:
Update travis config to something that works on macOS.

natebird committed:
Remove local tests in favor of using the Fluent tests.

natebird committed:
Update package name to PostgreSQLDriver

natebird committed:
Just test the Fluent tests on linux

natebird committed:
Major overhaul of the Driver and Serializer.

natebird committed:
Update readme for support of Swift 3.1 and new documentation location.

natebird committed:
ForeignKey + Logger updates

natebird committed:
Updates to help compile.

natebird committed:
Fixes for ForeignKey tests.

vzsg committed:
Ignore Package.pins

vzsg committed:
Replace backtick escaping with double quotes

PostgreSQL uses double quotes `""` to escape identifiers. Using
backticks (like MySQL) causes invalid SQL errors unconditionally.

vzsg committed:
Replace DATETIME type with TIMESTAMP

The former is not supported in Postgres.

vzsg committed:
Remove index definitions from CREATE TABLE

This is not supported by Postgres.

vzsg committed:
Fix foreign key test and indentation of test sources

[skip ci] as patching Fluent is required for FKs to work.

vzsg committed:
Fix INSERT INTO and LIMIT syntax

The Fluent tests still fail a big time, but it's improving steadily.
Next up: eliminating ? placeholders from filters.

vzsg committed:
Use internal counter for positional parameters

This allows the logic to be used in a lot more places without
duplicating code from the general serializer.

[skip ci] as Fluent is currently broken.

vzsg committed:
Eliminate MySQL strings, booleans and unsigned ints

[skip ci] as the tests are still broken.

natebird committed:
Added back in all the old serializer overrides updated to Fluent 2.0.

[skip ci] as the tests are still broken.

vzsg committed:
Remove the MySQL-specific extra types

vzsg committed:
Undo duplicate RETURNING id clause

vzsg committed:
Merge pull request #26 from natebird/2.0.0-beta

[WIP] 2.0.0 beta

ChasonTang commented:
It is seem that the tests are broken.The module name of `@testable import PostgreSQLDriverTests` is wrong

vzsg commented:
Where do you see that line? All tests have `import FluentPostgreSQL` where necessary in this branch. Anyhow, the tests currently fail due to missing parts.

One of the current problems is the missing support for timestamps from the wire protocol. It's a work in progress.

ChasonTang commented on Tests/LinuxMain.swift:
```diff
@@ -1,13 +1,10 @@
 
 import XCTest
-@testable import FluentPostgreSQLTests
+@testable import PostgreSQLDriverTests
```
PostgreSQLDriverTests is incorrect module name

ChasonTang replied:
@vzsg

vzsg replied:
Great catch, thanks.

ChasonTang replied:
Perhaps we can open an issue to present the progress that porting PostgreSQLDriver to vapor2

sroebert committed:
Renamed FluentPostgreSQL to PostgreSQLDriver to match other changes

vzsg committed:
Fix module and test class name for Linux

The tests are still broken, so [skip ci] to spare some time.

sroebert committed:
Updates for postgresql beta 4

sroebert committed:
Small reorder of parameters to match MySQL-Driver

vzsg committed:
Fix build after lower-level driver changes

vzsg committed:
Fix deleteIndex by removing override

`ALTER TABLE ... DROP INDEX` is not Postgres syntax.
The default works fine, however.

sroebert committed:
More renames (also in unit tests); Small reordering in PostgreSQLSerializer to match Fluent; Renamed ForeignKeys to Builder+PostgreSQL for more clarity

sroebert committed:
Merge branch 'vapor-2' of github.com:vapor-community/postgresql-driver into vapor-2

sroebert committed:
Renamed tests folder to make unit tests run again

vzsg committed:
Revert test module name change

There was nothing wrong with it.

Note to our future selves: the folder name matters, not our
interpretation of what it does in the background.

vzsg committed:
Kick CI
